server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -766,8 +766,8 @@ private ExternalReaderManager createReaderManager(RefreshWarmerListener external
                     DirectoryReader.open(indexWriter),
                     shardId
                 );
-                lastCommittedSegmentInfos = store.readLastCommittedSegmentsInfo();
```

Comment:
Unfortunately the change here was quite deliberate in that we need to access the `lastCommittedSegmentInfos` when creating the reader manager in stateless.

Can we instead fix the exception handling here?